### ISSUE TITLE
Fine Tuning does not work properly

### ISSUE THREAD

**Issue by abpani** [urchade/GLiNER#38]
input text: "Date of incorporation in Australia 27/11/1997 Must be in format dd/ mm / yyyy Website address ( URL ) Is the R&D entity controlled by one or more tax exempt entities ? Yes No confidential Raj"

labels = ["Person", "Date", "Organization", "Email", "Address"]

Finetuned  the model for 136 epochs with 8 batch size with almost 3000 training samples.
The model generates below output for a sample from training data itself.

[{'start': 20, 'end': 28, 'text': 'Number (', 'label': 'Person'},
 {'start': 49, 'end': 52, 'text': '9 6', 'label': 'Person'},
 {'start': 246, 'end': 248, 'text': 'mm', 'label': 'Date'},
 {'start': 274, 'end': 279, 'text': 'URL )', 'label': 'Organization'}]

**Comment by urchade** [comment]
Start and end is at the token/word level not character level. @abpani

**Comment by urchade** [comment]
I think the problem comes from tokenization

**Comment by imihalcea** [comment]
Hi,

I'm experiencing the same issue during fine-tuning. I have a data set that can be shared if needed.

**Comment by urchade** [comment]
The result is weird. Does the loss decrease during training ? Show an example of training data

**Comment by imihalcea** [comment]
{
'tokenized_text': ['new', 'appointment', 'set', 'for', 'cory', 'on', '04', '/', '04', '/', '1903', '.', 'arlo', ',', 'whom', 'resides', 'in', 'the', 'northeast', 'part', 'of', 'rowe', 'park', ',', 'is', 'to', 'start', 'session', 'at', '7', ':', '27', 'am', '.'],
 'ner': [[12, 12, 'firstname'], [6, 10, 'date'], [12, 12, 'firstname'], [18, 18, 'ordinaldirection'], [21, 22, 'street'], [29, 32, 'time']]
 },
 {
 'tokenized_text': ['hello', ',', 'our', 'next', 'counseling', 'session', 'will', 'be', 'held', 'at', 'the', 'northwest', 'part', 'of', 'the', 'school', '.', 'please', 'receive', 'payments', 'through', '0xc9a3f7a5edcfaf65f3e65a87e7bdecac8c51eb7b', '.', 'looking', 'forward', 'to', 'sharing', 'with', 'you', '.', 'stay', 'safe', ',', 'strategist', '.'], 'ner': [[11, 11, 'ordinaldirection'], [21, 21, 'ethereumaddress'], [33, 33, 'job']]}
 }
 
 @urchade here are 2 samples. My problem is that I have poor inference results when I evaluate the performance of the fine-tuned model. If don't see something obviously wrong in the dataset samples I'll come back with details about the dataset and loss evolution during training.
 
 Thank you .

**Comment by urchade** [comment]
the dataset seems to be in the correct format

**Comment by abpani** [comment]
Yes my dataset is also in same format.

**Comment by imihalcea** [comment]
@urchade the fine-tuning notebook is ok. After a closer look,  the poor performance came from an issue with my evaluation data set.  Thank's!

**Comment by urchade** [comment]
Does the performance improve upon standard NER architecture? (Ex BERT token classification)
